internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -173,10 +168,19 @@ func (h *PolicyChangeHandler) handleFleetServerHosts(ctx context.Context, c *con
 			h.config.Fleet.Client.Host = prevHost
 			h.config.Fleet.Client.Hosts = prevHosts
 			h.config.Fleet.Client.Transport.Proxy = prevProxy
+			if prevTLSNil {
+				h.config.Fleet.Client.Transport.TLS = nil
+			} else {
+				h.config.Fleet.Client.Transport.TLS.CAs = prevCAs
```

Comment:
Why not just save the entire TLS configuration instead of a boolean and individual fields from it?